app/src/main/java/org/oppia/android/app/application/AbstractOppiaApplication.kt

```diff
@@ -47,6 +51,16 @@ abstract class AbstractOppiaApplication(
     // TODO(#4751): Re-enable WorkManager for S+.
     if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S) {
       FirebaseApp.initializeApp(applicationContext)
+      val isDebugBuild = 0 != applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE
+      if (isDebugBuild) {
+        FirebaseAppCheck.getInstance().installAppCheckProviderFactory(
```

Comment:
Will this affect Firebase analytics, too?

Reply:
No, analytics has it's own seperate security and is not protected by Appcheck.

<img width="1000" alt="Screenshot 2024-01-26 at 09 15 24" src="https://github.com/oppia/oppia-android/assets/59600948/b7ad29c1-48a0-4f93-8bf8-e2b514489985">

Reply:
Got it--thanks! Could you maybe add a comment here to explain that? The code doesn't seem Firestore-specific. And, please also include what we would need to do in order to enable it for analytics if we wanted to in the future (or that it can't be if it can't).

Reply:
Added a comment explaining the usage of the library